lib/array.rb

```diff
@@ -0,0 +1,9 @@
+class Array
+
+  def new_inject (a=self.shift, &block)
+    self.each do |element|
+      a = block.call(a, element)
+    end
+    p a
```

Comment:
Don't put `puts` in production code!  You just need to return a, which you can do with `return a` or just `a` instead of `p a`